src/DefaultUser.php

```diff
@@ -31,10 +31,14 @@ final class DefaultUser implements UserInterface
     private $roles;
 
     /**
-     * @var array
+     * @psalm-var array<string|empty, mixed>
```

Comment:
Overall, `array<string|empty, mixed>` is equivalent to `array<string, mixed>` (the empty case is included into a map case)

Let's get rid of `empty` here, since it is redundant.

Reply:
To be clear: `array<empty, empty>|array<string, mixed>` is simplified down implicitly to `array<string, mixed>` at type level